documentation/Templates.md

```diff
@@ -67,6 +67,8 @@ Additionally to the [tags and filters](https://stencil.fuller.li/en/latest/built
 | Node | Description |
 | ---- | ----------- |
 | `{% set my_capture_variable %}`<br>`{% endset %}` | node pair allowing you to capture the rendering of nodes inside it, so you can reuse it later by reusing `{{ my_capture_variable }}`.<br>This is useful if you intend to reuse the same content at multiple places — like the `sceneID` is for the `storyboards-default.stencil` template for example. |
+| `{% func my_block a b c %}`<br>`{{args.a}} - {{args.b}} - {{args.c}}`<br>`{% endfunc %}` | node pair allowing you to define a reusable block of nodes inside (like a function), so you can reuse it later.<br>In contrast to the `set` tag, the capturing for `func` is lazy, it will only be processed when actually called. When processed, the block will provide a variable `args` containing all the parameter values.<br>Similar to the `for` tag, this tag provides it's own context scope when called, so variables inside of it are not available outside of it. |
```

Comment:
I wonder if it might instead make sense to make use of `{% include %}` tag. You could break component into separate files and make them more reusable/composable.

Short Example:

```html+django
{% for box in boxes %}
    {% include "render_box.html" %}
{% endfor %}
```

This has the benefit of allowing users to override default reusable components without having to overriding complete templates.

Example:

```swift
FileSystemLoader(paths: ["~/Projects/some/user/templates", "/usr/local/share/swiftgen/templates"])
```

Where these paths look as follows:

```
swiftgen/templates
├── article_detail.html
├── article_list.html
├── comment_detail.html

user/templates
└── comment_detail.html
```

Where the user has only overridden `comment_detail.html` to be the following:

##### `user/templates/comment_detail.html`

```html+django
<li>{{ comment.author }} said {{ comment }}</li>
```

##### `swiftenv/templates/comment_detail.html`

```html+django
<li>
  <h5>{{ comment.author }} at {{ comment.published }}</h5>
  {{ comment|markdown }}
</li>
```

##### `article_detail.html`

```html+django
<h3>{{ article.title }}</h3>

{{ article|markdown }}

<h4>Comments</h4>

<ul>
  {% for comment in article.comments %}
    {% include "comment_detail.html" %}
  {% endfor %}
</ul>
```

Now, when `article_detail.html` is rendered, the users overridden `comment_detail.html` will be used for each comment.